### ISSUE TITLE
Jinja2: UnicodeDecodeError with RichText in StreamField

### ISSUE THREAD

**Issue by m1kola** [torchbox/wagtail#2057]
Hello!

I am getting `UnicodeDecodeError` while rendering `post.body` when I put unicode (cyrillic) characters to RichText in StreamField. Meanwhile in admin interface everything is ok. 
```
UnicodeDecodeError: 'ascii' codec can't decode byte 0xd0 in position 55: ordinal not in range(128)
```

Seems like broken serialization. `post.body` in database (PostgreSQL 9.4.5):
```
select * from blog_blogpage where page_ptr_id = 5;
-[ RECORD 1 ]-+--------------------------------------------------------------------------------------------------
page_ptr_id   | 5
subtitle      | Саб тайтл тест
body          | [{"type": "rich_text", "value": "<p>\u0411\u0430\u0434\u0438 \u0442\u0435\u0441\u0442<br/></p>"}]
lead          | <p>Лид тест</p>
main_image_id | 1
```

If body just RichTextField, everything works fine.

I can reproduce this on following versions of wagtail: 1.2, 1.3rc1

**Comment by kaedroho** [comment]
The serialisation looks OK to me. JSON normally escapes unicode characters.

It looks looks like Python's trying to decode the bytes into ASCII text which doesn't support Unicode characters. So this is most likely Python 2 specific. Could you provide a traceback?

**Comment by m1kola** [comment]
@kaedroho, I think you're right. I ran my application with Python 2.7.10. I'll send you traceback when it will be possible.

**Comment by m1kola** [comment]
@kaedroho, here is traceback.

```
Traceback:

File "/Users/m1kola/.virtualenvs/blog/lib/python2.7/site-packages/django/core/handlers/base.py" in get_response
  174.                     response = self.process_exception_by_middleware(e, request)

File "/Users/m1kola/.virtualenvs/blog/lib/python2.7/site-packages/django/core/handlers/base.py" in get_response
  172.                     response = response.render()

File "/Users/m1kola/.virtualenvs/blog/lib/python2.7/site-packages/django/template/response.py" in render
  160.             self.content = self.rendered_content

File "/Users/m1kola/.virtualenvs/blog/lib/python2.7/site-packages/django/template/response.py" in rendered_content
  137.         content = template.render(context, self._request)

File "/Users/m1kola/.virtualenvs/blog/lib/python2.7/site-packages/django/template/backends/jinja2.py" in render
  70.         return self.template.render(context)

File "/Users/m1kola/.virtualenvs/blog/lib/python2.7/site-packages/jinja2/environment.py" in render
  989.         return self.environment.handle_exception(exc_info, True)

File "/Users/m1kola/.virtualenvs/blog/lib/python2.7/site-packages/jinja2/environment.py" in handle_exception
  754.         reraise(exc_type, exc_value, tb)

File "/Users/m1kola/Projects/blog_mikola_by/src/blog/jinja2/blog/blog_page.html" in top-level template code
  2. {% import 'blog/macro/post.html' as post_macro %}

File "/Users/m1kola/Projects/blog_mikola_by/src/base/jinja2/base/_base.html" in top-level template code
  58.         {% block main_content %}{% endblock %}

File "/Users/m1kola/Projects/blog_mikola_by/src/blog/jinja2/blog/blog_page.html" in block "main_content"
  6.     {{ post_macro.article(page) }}

File "/Users/m1kola/Projects/blog_mikola_by/src/blog/jinja2/blog/macro/post.html" in template
  24.                     {{ post.body }}

Exception Type: UnicodeDecodeError at /en/test/
Exception Value: 'ascii' codec can't decode byte 0xd0 in position 55: ordinal not in range(128)
```

Maybe `pip freeze` will be useful.

```
beautifulsoup4==4.4.1
Django==1.9
django-appconf==1.0.1
django-compressor==1.6
django-debug-toolbar==1.4
django-modelcluster==1.1
django-taggit==0.17.6
django-treebeard==3.0
djangorestframework==3.3.2
html5lib==0.9999999
Jinja2==2.8
MarkupSafe==0.23
Pillow==2.6.1
psycopg2==2.6.1
Pygments==2.0.2
pytz==2015.7
six==1.10.0
sqlparse==0.1.18
Unidecode==0.4.18
uWSGI==2.0.11.2
wagtail==1.3rc1
Willow==0.2.2
```

**Comment by m1kola** [comment]
I can't reproduce this exception with Django templates on `wagtaildemo` project.

**Comment by m1kola** [comment]
Fixed in #2125
